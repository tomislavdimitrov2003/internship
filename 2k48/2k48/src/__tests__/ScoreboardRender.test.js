import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import React from 'react';
import RenderScoreboard from "../ScoreboardRenderer";


it('renders scoreboard correctly', () => {
    const rows = [['a', 50], ['b', 60]];
    const wrapper = shallow(<RenderScoreboard rows={rows}/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('sort scoreboard by name correctly', () => {
    const rows = [['a', 50], ['b', 60]];
    const wrapper = shallow(<RenderScoreboard rows={rows}/>);
    wrapper.find('Score').simulate('click');
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('sort scoreboard by score correctly', () => {
    const rows = [['a', 50], ['b', 60]];
    const wrapper = shallow(<RenderScoreboard rows={rows}/>);
    wrapper.find('Score').simulate('click');
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});