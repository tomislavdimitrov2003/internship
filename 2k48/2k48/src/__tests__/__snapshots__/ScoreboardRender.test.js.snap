// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders scoreboard correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-2"
  >
    <EnhancedTableToolbar
      numSelected={0}
    />
    <WithStyles(ForwardRef(TableContainer))>
      <WithStyles(ForwardRef(Table))
        aria-label="enhanced table"
        aria-labelledby="tableTitle"
        className="makeStyles-table-3"
        size="medium"
      >
        <EnhancedTableHead
          classes={
            Object {
              "paper": "makeStyles-paper-2",
              "root": "makeStyles-root-1",
              "table": "makeStyles-table-3",
              "visuallyHidden": "makeStyles-visuallyHidden-4",
            }
          }
          numSelected={0}
          onRequestSort={[Function]}
          order="desc"
          orderBy="score"
          rowCount={2}
        />
        <WithStyles(ForwardRef(TableBody))>
          <WithStyles(ForwardRef(TableRow))
            role="checkbox"
            tabIndex={-1}
          >
            <WithStyles(ForwardRef(TableCell))
              padding="checkbox"
            />
            <WithStyles(ForwardRef(TableCell))
              component="th"
              id="enhanced-table-checkbox-0"
              padding="none"
              scope="row"
            />
            <WithStyles(ForwardRef(TableCell))
              align="right"
            />
          </WithStyles(ForwardRef(TableRow))>
          <WithStyles(ForwardRef(TableRow))
            role="checkbox"
            tabIndex={-1}
          >
            <WithStyles(ForwardRef(TableCell))
              padding="checkbox"
            />
            <WithStyles(ForwardRef(TableCell))
              component="th"
              id="enhanced-table-checkbox-1"
              padding="none"
              scope="row"
            />
            <WithStyles(ForwardRef(TableCell))
              align="right"
            />
          </WithStyles(ForwardRef(TableRow))>
          <WithStyles(ForwardRef(TableRow))
            style={
              Object {
                "height": 159,
              }
            }
          >
            <WithStyles(ForwardRef(TableCell))
              colSpan={6}
            />
          </WithStyles(ForwardRef(TableRow))>
        </WithStyles(ForwardRef(TableBody))>
      </WithStyles(ForwardRef(Table))>
    </WithStyles(ForwardRef(TableContainer))>
    <WithStyles(ForwardRef(TablePagination))
      component="div"
      count={2}
      onPageChange={[Function]}
      onRowsPerPageChange={[Function]}
      page={0}
      rowsPerPage={5}
      rowsPerPageOptions={
        Array [
          5,
          10,
          25,
        ]
      }
    />
  </WithStyles(ForwardRef(Paper))>
  <WithStyles(ForwardRef(FormControlLabel))
    control={
      <WithStyles(ForwardRef(Switch))
        checked={false}
        onChange={[Function]}
      />
    }
    label="Dense padding"
  />
</div>
`;
